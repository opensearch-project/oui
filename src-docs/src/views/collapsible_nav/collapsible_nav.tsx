/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React, { useState } from 'react';

import { OuiCollapsibleNav } from '../../../../src/components/collapsible_nav';
import { OuiButton } from '../../../../src/components/button';
import { OuiTitle } from '../../../../src/components/title';
import { OuiSpacer } from '../../../../src/components/spacer';
import { OuiText } from '../../../../src/components/text';
import { OuiCode } from '../../../../src/components/code';

export default () => {
  const [navIsOpen, setNavIsOpen] = useState<boolean>(
    JSON.parse(
      String(localStorage.getItem('ouiCollapsibleNavExample--isDocked'))
    ) || false
  );
  const [navIsDocked, setNavIsDocked] = useState<boolean>(
    JSON.parse(
      String(localStorage.getItem('ouiCollapsibleNavExample--isDocked'))
    ) || false
  );

  return (
    <>
      <OuiCollapsibleNav
        isOpen={navIsOpen}
        isDocked={navIsDocked}
        size={240}
        button={
          <OuiButton onClick={() => setNavIsOpen((isOpen) => !isOpen)}>
            Toggle nav
          </OuiButton>
        }
        onClose={() => setNavIsOpen(false)}>
        <div style={{ padding: 16 }}>
          <OuiTitle>
            <h2>Navigation</h2>
          </OuiTitle>
          <OuiSpacer />
          <OuiText size="s" color="subdued">
            <p>
              The docked status is being stored in{' '}
              <OuiCode>localStorage</OuiCode>.
            </p>
          </OuiText>
          <OuiSpacer />
          <OuiButton
            onClick={() => {
              setNavIsDocked(!navIsDocked);
              localStorage.setItem(
                'ouiCollapsibleNavExample--isDocked',
                JSON.stringify(!navIsDocked)
              );
            }}>
            Docked: {navIsDocked ? 'on' : 'off'}
          </OuiButton>
        </div>
      </OuiCollapsibleNav>

      {navIsDocked && (
        <OuiText size="s" color="subdued">
          <p>
            The <OuiCode>button</OuiCode> gets hidden by default when the nav is
            docked unless you set{' '}
            <OuiCode language="js">showButtonIfDocked = true</OuiCode>.
          </p>
        </OuiText>
      )}
    </>
  );
};
