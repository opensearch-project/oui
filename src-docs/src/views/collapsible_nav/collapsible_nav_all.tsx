/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React, { useState } from 'react';

import {
  OuiCollapsibleNav,
  OuiCollapsibleNavGroup,
} from '../../../../src/components/collapsible_nav';
import {
  OuiHeaderSectionItemButton,
  OuiHeaderLogo,
  OuiHeader,
} from '../../../../src/components/header';
import { OuiIcon } from '../../../../src/components/icon';
import { OuiButtonEmpty } from '../../../../src/components/button';
import { OuiPageTemplate } from '../../../../src/components/page';
import {
  OuiListGroupItem,
  OuiListGroup,
} from '../../../../src/components/list_group';
import { OuiFlexItem } from '../../../../src/components/flex';

import {
  ManagementLinks,
  OpenSearchDashboardsLinks,
  OpenSearchPluginLinks,
} from './collapsible_nav_list';
import { OuiShowFor } from '../../../../src/components/responsive';
import { OuiImage } from '../../../../src/components/image';
import contentSvg from '../../images/content.svg';
import { useExitPath } from '../../services/routing/routing';

const CollapsibleNavAll = () => {
  const exitPath = useExitPath();
  const [navIsOpen, setNavIsOpen] = useState(true);
  const [navIsDocked, setNavIsDocked] = useState(
    JSON.parse(String(localStorage.getItem('nav2IsDocked'))) || false
  );

  /**
   * Accordion toggling
   */
  const [openGroups, setOpenGroups] = useState(
    JSON.parse(String(localStorage.getItem('openNavGroups'))) || [
      'OpenSearch Dashboards',
      'OpenSearch Plugins',
      'Management',
    ]
  );

  // Save which groups are open and which are not with state and local store
  const toggleAccordion = (isOpen: boolean, title?: string) => {
    if (!title) return;
    const itExists = openGroups.includes(title);
    if (isOpen) {
      if (itExists) return;
      openGroups.push(title);
    } else {
      const index = openGroups.indexOf(title);
      if (index > -1) {
        openGroups.splice(index, 1);
      }
    }
    setOpenGroups([...openGroups]);
    localStorage.setItem('openNavGroups', JSON.stringify(openGroups));
  };

  const collapsibleNav = (
    <OuiCollapsibleNav
      id="guideCollapsibleNavAllExampleNav"
      aria-label="Main navigation"
      isOpen={navIsOpen}
      isDocked={navIsDocked}
      button={
        <OuiHeaderSectionItemButton
          aria-label="Toggle main navigation"
          onClick={() => setNavIsOpen(!navIsOpen)}>
          <OuiIcon type={'menu'} size="m" aria-hidden="true" />
        </OuiHeaderSectionItemButton>
      }
      onClose={() => setNavIsOpen(false)}>
      {/* BOTTOM */}
      <OuiFlexItem className="oui-yScroll">
        {/* OpenSearch Dashboards section */}
        <OuiCollapsibleNavGroup
          title="OpenSearch Dashboards"
          iconType="logoOpenSearch"
          isCollapsible={true}
          initialIsOpen={openGroups.includes('OpenSearch Dashboards')}
          onToggle={(isOpen: boolean) =>
            toggleAccordion(isOpen, 'OpenSearch Dashboards')
          }>
          <OuiListGroup
            aria-label="OpenSearch Dashboards" // A11y : OuiCollapsibleNavGroup can't correctly pass the `title` as the `aria-label` to the right HTML element, so it must be added manually
            listItems={OpenSearchDashboardsLinks}
            maxWidth="none"
            color="subdued"
            gutterSize="none"
            size="s"
          />
        </OuiCollapsibleNavGroup>
        <OuiCollapsibleNavGroup
          title="OpenSearch Plugins"
          isCollapsible={true}
          initialIsOpen={openGroups.includes('OpenSearch Plugins')}
          onToggle={(isOpen: boolean) =>
            toggleAccordion(isOpen, 'OpenSearch Plugins')
          }>
          <OuiListGroup
            aria-label="OpenSearch Plugins" // A11y : OuiCollapsibleNavGroup can't correctly pass the `title` as the `aria-label` to the right HTML element, so it must be added manually
            listItems={OpenSearchPluginLinks}
            maxWidth="none"
            color="subdued"
            gutterSize="none"
            size="s"
          />
        </OuiCollapsibleNavGroup>

        <OuiCollapsibleNavGroup
          title="Management"
          isCollapsible={true}
          initialIsOpen={openGroups.includes('Management')}
          onToggle={(isOpen: boolean) => toggleAccordion(isOpen, 'Management')}>
          <OuiListGroup
            aria-label="Management" // A11y : OuiCollapsibleNavGroup can't correctly pass the `title` as the `aria-label` to the right HTML element, so it must be added manually
            listItems={ManagementLinks}
            maxWidth="none"
            color="subdued"
            gutterSize="none"
            size="s"
          />
        </OuiCollapsibleNavGroup>

        {/* Docking button only for larger screens that can support it*/}
        <OuiShowFor sizes={['l', 'xl']}>
          <OuiCollapsibleNavGroup>
            <OuiListGroupItem
              size="xs"
              color="subdued"
              label={`${navIsDocked ? 'Undock' : 'Dock'} navigation`}
              onClick={() => {
                setNavIsDocked(!navIsDocked);
                localStorage.setItem(
                  'nav2IsDocked',
                  JSON.stringify(!navIsDocked)
                );
              }}
              iconType={navIsDocked ? 'lock' : 'lockOpen'}
            />
          </OuiCollapsibleNavGroup>
        </OuiShowFor>
      </OuiFlexItem>
    </OuiCollapsibleNav>
  );

  const leftSectionItems = [
    collapsibleNav,
    <OuiHeaderLogo href={exitPath} iconType="logoOpenSearch">
      OpenSearch UI
    </OuiHeaderLogo>,
  ];

  return (
    <>
      <OuiHeader
        position="fixed"
        sections={[
          {
            items: leftSectionItems,
            borders: 'right',
          },
          {
            items: [
              <OuiButtonEmpty href={exitPath} iconType="exit">
                Exit full screen
              </OuiButtonEmpty>,
            ],
          },
        ]}
      />

      <OuiPageTemplate template="centeredBody">
        <OuiImage size="fullWidth" alt="Fake paragraph" url={contentSvg} />
      </OuiPageTemplate>
    </>
  );
};

export default CollapsibleNavAll;
