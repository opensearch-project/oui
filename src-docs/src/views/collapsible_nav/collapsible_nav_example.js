/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';
import { Link } from 'react-router-dom';

import { GuideSectionTypes } from '../../components';

import {
  OuiCode,
  OuiCollapsibleNav,
  OuiText,
  OuiCallOut,
  OuiCollapsibleNavGroup,
  OuiHorizontalRule,
} from '../../../../src/components';

import { collapsibleNavConfig } from './playground';

import CollapsibleNav from './collapsible_nav';
const collapsibleNavSource = require('!!raw-loader!./collapsible_nav');

import CollapsibleNavGroup from './collapsible_nav_group';
const collapsibleNavGroupSource = require('!!raw-loader!./collapsible_nav_group');

import CollapsibleNavList from './collapsible_nav_list';
const collapsibleNavListSource = require('!!raw-loader!./collapsible_nav_list');

import CollapsibleNavAll from './collapsible_nav_all';
const collapsibleNavAllSource = require('!!raw-loader!./collapsible_nav_all');

export const CollapsibleNavExample = {
  title: 'Collapsible nav',
  intro: (
    <OuiText>
      <p>
        This is a high level component that creates a flyout-style navigational
        pane.
      </p>
    </OuiText>
  ),
  sections: [
    {
      source: [
        {
          type: GuideSectionTypes.JS,
          code: collapsibleNavSource,
        },
      ],
      text: (
        <>
          <p>
            <strong>OuiCollapsibleNav</strong> is a custom implementation of{' '}
            <Link to="/layout/flyout">
              <strong>OuiFlyout</strong>
            </Link>
            ; the visibility of which must still be maintained by the consuming
            application. An extra feature that it provides is the ability to{' '}
            <OuiCode>dock</OuiCode> the flyout. This affixes the flyout to the
            window and pushes the body content by adding left side padding.
          </p>
          <OuiCallOut
            iconType="tableOfContents"
            title="Docking is not possible on small screens because it would force less real estate for the page content."
          />
        </>
      ),
      props: { OuiCollapsibleNav },
      demo: <CollapsibleNav />,
      snippet: `<OuiCollapsibleNav
  size={240}
  button={<OuiButton onClick={() => setNavIsOpen(!navIsOpen)}>Toggle nav</OuiButton>}
  isOpen={navIsOpen}
  isDocked={navIsDocked}
  onClose={() => setNavIsOpen(false)}
/>`,
      playground: collapsibleNavConfig,
    },
    {
      title: 'Collapsible nav group',
      source: [
        {
          type: GuideSectionTypes.JS,
          code: collapsibleNavGroupSource,
        },
      ],
      text: (
        <>
          <p>
            An <strong>OuiCollapsibleNavGroup</strong> adds some basic borders
            and <OuiCode>background</OuiCode> color of <OuiCode>none</OuiCode>,{' '}
            <OuiCode>light</OuiCode>, or <OuiCode>dark</OuiCode>. Give each
            section a heading by providing an optional <OuiCode>title</OuiCode>{' '}
            and <OuiCode>iconType</OuiCode>. Make the section collapsible (
            <Link to="/layout/accordion">accordion style</Link>) with{' '}
            <OuiCode language="js">isCollapsible=true</OuiCode>.
          </p>
          <p>
            When in <OuiCode>isCollapsible</OuiCode> mode, a{' '}
            <OuiCode>title</OuiCode> and{' '}
            <OuiCode language="ts">initialIsOpen:boolean</OuiCode> is required.
          </p>
        </>
      ),
      props: {
        OuiCollapsibleNavGroup,
      },
      demo: <CollapsibleNavGroup />,
      demoPanelProps: {
        paddingSize: 'none',
      },
      snippet: `<OuiCollapsibleNavGroup
  title="Nav group"
  iconType="logo"
  isCollapsible={true}
  initialIsOpen={true}
  background="none"
/>`,
    },
    {
      title: 'Nav groups with lists and other content',
      source: [
        {
          type: GuideSectionTypes.JS,
          code: collapsibleNavListSource,
        },
      ],
      text: (
        <>
          <p>
            <strong>OuiCollapsibleNavGroups</strong> can contain any children.
            They work well with{' '}
            <Link to="/display/list-group">
              <strong>OuiListGroup, OuiPinnableListGroup</strong>
            </Link>{' '}
            and simple{' '}
            <Link to="/navigation/link">
              <strong>OuiText</strong>
            </Link>
            .
          </p>
          <p>Below are a few established patterns to use.</p>
        </>
      ),
      demo: (
        <div>
          <CollapsibleNavList />
          <OuiHorizontalRule margin="none" />
        </div>
      ),
      demoPanelProps: {
        paddingSize: 'none',
      },
      snippet: `<OuiCollapsibleNavGroup
  title="OpenSearch Dashboards"
  iconType="logoOpenSearch"
  isCollapsible={true}
  initialIsOpen={true}>
  <OuiPinnableListGroup
    aria-label="OpenSearch Dashboards"
    listItems={[
      { label: 'Discover' },
      { label: 'Visualize' }
    ]}
    onPinClick={() => {}}
    maxWidth="none"
    color="subdued"
    gutterSize="none"
    size="s"
  />
</OuiCollapsibleNavGroup>`,
    },
    {
      title: 'Full pattern with header',
      source: [
        {
          type: GuideSectionTypes.JS,
          code: collapsibleNavAllSource,
        },
      ],
      text: (
        <>
          <h3>Putting it all together</h3>
          <p>
            The button below will launch a full screen example that includes{' '}
            <Link to="/layout/header">
              <strong>OuiHeader</strong>
            </Link>{' '}
            with a toggle button to open an <strong>OuiCollapsibleNav</strong>.
            The contents of which are multiple{' '}
            <strong>OuiCollapsibleNavGroups</strong>
          </p>
          <p>
            This is just a pattern and should be treated as such. Consuming
            applications will need to create the navigation groups according to
            their context and save the states as is appropriate to their data
            store.
          </p>
        </>
      ),
      fullScreen: {
        slug: 'collapsible-nav-all',
        demo: <CollapsibleNavAll />,
      },
    },
  ],
};
