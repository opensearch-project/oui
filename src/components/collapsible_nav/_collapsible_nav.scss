/*!
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

// Extends <OuiFlyout>

.ouiCollapsibleNav:not([class*='push']) {
  z-index: $ouiZNavigation !important; // sass-lint:disable-line no-important
}

.ouiCollapsibleNav.ouiFlyout {
  background-color: $ouiCollapsibleNavBackgroundColor;
}
