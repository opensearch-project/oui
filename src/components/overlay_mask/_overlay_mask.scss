/*!
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

.ouiOverlayMask {
  position: fixed;
  inset: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  padding-bottom: 10vh;
  animation: ouiAnimFadeIn $ouiAnimSpeedFast ease-in;

  $backgroundColor: $ouiColorEmptyShade;

  @if  lightness($ouiTextColor) > 50 {
    $backgroundColor: $ouiColorLightShade;
  }

  background: transparentize($backgroundColor, .2);
}

.ouiBody-hasOverlayMask {
  overflow: hidden;
}

// Handling the z-index based on whether it should be displayed above or below the header
.ouiOverlayMask--aboveHeader {
  z-index: $ouiZMask;
}

.ouiOverlayMask--belowHeader {
  z-index: $ouiZMaskBelowHeader;
}
