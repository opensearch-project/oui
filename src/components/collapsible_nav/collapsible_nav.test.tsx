/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import { render } from '@testing-library/react';
import { requiredProps } from '../../test';

import { OuiCollapsibleNav } from './collapsible_nav';
import { OuiOverlayMaskProps } from '../overlay_mask';

jest.mock('../overlay_mask', () => ({
  OuiOverlayMask: ({ headerZindexLocation, ...props }: any) => (
    <div {...props} />
  ),
}));

jest.mock('../portal', () => ({
  OuiPortal: ({ children }: { children: any }) => children,
}));

const propsNeededToRender = { id: 'id', isOpen: true, onClose: () => {} };
const flyoutProps = {
  size: 240,
  ownFocus: false,
  outsideClickCloses: false,
  maskProps: { headerZindexLocation: 'above' } as OuiOverlayMaskProps,
};

describe('OuiCollapsibleNav', () => {
  test('is rendered', () => {
    render(<OuiCollapsibleNav {...propsNeededToRender} {...requiredProps} />);

    expect(document.body).toMatchSnapshot();
  });

  describe('props', () => {
    test('onClose', () => {
      render(<OuiCollapsibleNav {...propsNeededToRender} onClose={() => {}} />);

      expect(document.body).toMatchSnapshot();
    });

    test('size', () => {
      render(<OuiCollapsibleNav {...propsNeededToRender} size={240} />);

      expect(document.body).toMatchSnapshot();
    });

    test('isDocked', () => {
      const { container } = render(
        <OuiCollapsibleNav {...propsNeededToRender} isDocked={true} />
      );

      expect(container).toMatchSnapshot();
    });

    test('dockedBreakpoint', () => {
      render(
        <OuiCollapsibleNav {...propsNeededToRender} dockedBreakpoint={500} />
      );

      expect(document.body).toMatchSnapshot();
    });

    test('button', () => {
      render(
        <OuiCollapsibleNav {...propsNeededToRender} button={<button />} />
      );

      expect(document.body).toMatchSnapshot();
    });

    test('showButtonIfDocked', () => {
      render(
        <OuiCollapsibleNav
          {...propsNeededToRender}
          button={<button />}
          isDocked={true}
          showButtonIfDocked={true}
        />
      );

      expect(document.body).toMatchSnapshot();
    });

    test('accepts OuiFlyout props', () => {
      render(<OuiCollapsibleNav {...propsNeededToRender} {...flyoutProps} />);

      expect(document.body).toMatchSnapshot();
    });
  });

  describe('close button', () => {
    test('can be hidden', () => {
      render(
        <OuiCollapsibleNav {...propsNeededToRender} hideCloseButton={true} />
      );

      expect(document.body).toMatchSnapshot();
    });
  });

  test('does not render if isOpen is false', () => {
    const { container } = render(
      <OuiCollapsibleNav onClose={() => {}} id="id" />
    );

    expect(container).toMatchSnapshot();
  });
});
