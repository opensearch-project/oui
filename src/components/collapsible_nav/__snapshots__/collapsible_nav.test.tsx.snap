// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OuiCollapsibleNav close button can be hidden 1`] = `
[
  <div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <nav
        class="ouiFlyout ouiFlyout--left ouiFlyout--paddingNone ouiCollapsibleNav"
        id="id"
        style="width: 320px;"
        tabindex="-1"
      >
        <button
          aria-label="Close this dialog"
          class="ouiButtonIcon ouiButtonIcon--text ouiButtonIcon--fill ouiButtonIcon--xSmall ouiFlyout__closeButton ouiFlyout__closeButton--outside"
          data-test-subj="ouiFlyoutCloseButton"
          type="button"
        >
          <span
            aria-hidden="true"
            class="ouiButtonIcon__icon"
            color="inherit"
            data-ouiicon-type="cross"
          />
        </button>
      </nav>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>,
]
`;

exports[`OuiCollapsibleNav does not render if isOpen is false 1`] = `null`;

exports[`OuiCollapsibleNav is rendered 1`] = `
[
  <div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <nav
        aria-label="aria-label"
        class="ouiFlyout ouiFlyout--left ouiFlyout--paddingNone ouiCollapsibleNav testClass1 testClass2"
        data-test-subj="test subject string"
        id="id"
        style="width: 320px;"
        tabindex="-1"
      >
        <button
          aria-label="Close this dialog"
          class="ouiButtonIcon ouiButtonIcon--text ouiButtonIcon--fill ouiButtonIcon--xSmall ouiFlyout__closeButton ouiFlyout__closeButton--outside"
          data-test-subj="ouiFlyoutCloseButton"
          type="button"
        >
          <span
            aria-hidden="true"
            class="ouiButtonIcon__icon"
            color="inherit"
            data-ouiicon-type="cross"
          />
        </button>
      </nav>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>,
]
`;

exports[`OuiCollapsibleNav props accepts OuiFlyout props 1`] = `
[
  <div
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />,
  <div
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="1"
  />,
  <div
    data-focus-lock-disabled="false"
  >
    <nav
      class="ouiFlyout ouiFlyout--left ouiFlyout--paddingNone ouiCollapsibleNav"
      id="id"
      style="width: 240px;"
      tabindex="-1"
    >
      <button
        aria-label="Close this dialog"
        class="ouiButtonIcon ouiButtonIcon--text ouiButtonIcon--fill ouiButtonIcon--xSmall ouiFlyout__closeButton ouiFlyout__closeButton--outside"
        data-test-subj="ouiFlyoutCloseButton"
        type="button"
      >
        <span
          aria-hidden="true"
          class="ouiButtonIcon__icon"
          color="inherit"
          data-ouiicon-type="cross"
        />
      </button>
    </nav>
  </div>,
  <div
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />,
]
`;

exports[`OuiCollapsibleNav props button 1`] = `
[
  <button
    aria-controls="id"
    aria-expanded="true"
    aria-pressed="true"
  />,
  <div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <nav
        class="ouiFlyout ouiFlyout--left ouiFlyout--paddingNone ouiCollapsibleNav"
        id="id"
        style="width: 320px;"
        tabindex="-1"
      >
        <button
          aria-label="Close this dialog"
          class="ouiButtonIcon ouiButtonIcon--text ouiButtonIcon--fill ouiButtonIcon--xSmall ouiFlyout__closeButton ouiFlyout__closeButton--outside"
          data-test-subj="ouiFlyoutCloseButton"
          type="button"
        >
          <span
            aria-hidden="true"
            class="ouiButtonIcon__icon"
            color="inherit"
            data-ouiicon-type="cross"
          />
        </button>
      </nav>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>,
]
`;

exports[`OuiCollapsibleNav props dockedBreakpoint 1`] = `
[
  <div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <nav
        class="ouiFlyout ouiFlyout--left ouiFlyout--paddingNone ouiCollapsibleNav"
        id="id"
        style="width: 320px;"
        tabindex="-1"
      >
        <button
          aria-label="Close this dialog"
          class="ouiButtonIcon ouiButtonIcon--text ouiButtonIcon--fill ouiButtonIcon--xSmall ouiFlyout__closeButton ouiFlyout__closeButton--outside"
          data-test-subj="ouiFlyoutCloseButton"
          type="button"
        >
          <span
            aria-hidden="true"
            class="ouiButtonIcon__icon"
            color="inherit"
            data-ouiicon-type="cross"
          />
        </button>
      </nav>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>,
]
`;

exports[`OuiCollapsibleNav props isDocked 1`] = `
[
  <div
    data-focus-guard="true"
    style="width:1px;height:0px;padding:0;overflow:hidden;position:fixed;top:1px;left:1px"
    tabindex="-1"
  />,
  <div
    data-focus-guard="true"
    style="width:1px;height:0px;padding:0;overflow:hidden;position:fixed;top:1px;left:1px"
    tabindex="-1"
  />,
  <div
    data-focus-lock-disabled="disabled"
  >
    <nav
      class="ouiFlyout ouiFlyout--push ouiFlyout--left ouiFlyout--paddingNone ouiCollapsibleNav"
      id="id"
      style="width:320px"
      tabindex="-1"
    />
  </div>,
  <div
    data-focus-guard="true"
    style="width:1px;height:0px;padding:0;overflow:hidden;position:fixed;top:1px;left:1px"
    tabindex="-1"
  />,
]
`;

exports[`OuiCollapsibleNav props onClose 1`] = `
[
  <div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <nav
        class="ouiFlyout ouiFlyout--left ouiFlyout--paddingNone ouiCollapsibleNav"
        id="id"
        style="width: 320px;"
        tabindex="-1"
      >
        <button
          aria-label="Close this dialog"
          class="ouiButtonIcon ouiButtonIcon--text ouiButtonIcon--fill ouiButtonIcon--xSmall ouiFlyout__closeButton ouiFlyout__closeButton--outside"
          data-test-subj="ouiFlyoutCloseButton"
          type="button"
        >
          <span
            aria-hidden="true"
            class="ouiButtonIcon__icon"
            color="inherit"
            data-ouiicon-type="cross"
          />
        </button>
      </nav>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>,
]
`;

exports[`OuiCollapsibleNav props showButtonIfDocked 1`] = `
[
  <button
    aria-controls="id"
    aria-expanded="true"
    aria-pressed="true"
  />,
  <div
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="-1"
  />,
  <div
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="-1"
  />,
  <div
    data-focus-lock-disabled="disabled"
  >
    <nav
      class="ouiFlyout ouiFlyout--push ouiFlyout--left ouiFlyout--paddingNone ouiCollapsibleNav"
      id="id"
      style="width: 320px;"
      tabindex="-1"
    />
  </div>,
  <div
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="-1"
  />,
]
`;

exports[`OuiCollapsibleNav props size 1`] = `
[
  <div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <nav
        class="ouiFlyout ouiFlyout--left ouiFlyout--paddingNone ouiCollapsibleNav"
        id="id"
        style="width: 240px;"
        tabindex="-1"
      >
        <button
          aria-label="Close this dialog"
          class="ouiButtonIcon ouiButtonIcon--text ouiButtonIcon--fill ouiButtonIcon--xSmall ouiFlyout__closeButton ouiFlyout__closeButton--outside"
          data-test-subj="ouiFlyoutCloseButton"
          type="button"
        >
          <span
            aria-hidden="true"
            class="ouiButtonIcon__icon"
            color="inherit"
            data-ouiicon-type="cross"
          />
        </button>
      </nav>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>,
]
`;
